src/main/java/org/jitsi/impl/protocol/xmpp/ChatRoomImpl.java

```diff
@@ -58,7 +58,8 @@
      * Caches early presence packets triggered by Smack, before there was member
      * joined event.
      */
-    private Map<String, Presence> presenceCache = new HashMap<>();
+    private final Map<String, Presence> presenceCache
+        = new ConcurrentHashMap<>();
```

Comment:
I'm not sure that it's much better than before. It's true that the presenceCache will be consistently modified as far as the generic Map interface goes. But will it be consistently modified as far as ChatRoomImpl goes? I'm not sure. In 2 out of 3 uses presenceCache is protected by members, in the remaining 1 out of 3 uses it's not protected by members. Murky business.